### PR TITLE
upgpkg: hipblas 5.2.0-1

### LINKED ISSUE #800: [hipblas] rocblas.h instead of rocblas/rocblas.h
Attempting to install ROCm from the AUR without arch4edu and receiving a few build errors. Because the errors appear to not be linked, I'm creating two issues (#801). 

The first error is hopefully an easy fix, I'm receiving a build error for hipblas:
```
/home/whisper/.cache/paru/clone/hipblas/src/hipBLAS-rocm-5.1.3/library/src/hcc_detail/hipblas.cpp:7:10: fatal error: 'rocblas.h' file not found
#include "rocblas.h"
         ^~~~~~~~~~~
```

The [hipblas](https://github.com/ROCmSoftwarePlatform/hipBLAS/blob/develop/library/src/hcc_detail/hipblas.cpp) repository upstream appears to have this fixed, but the files from the AUR do not. 

In addition, is there a way to build a modified package and have it be accepted by the paru helper? Editing the file directly from the paru cache and executing `make install` in the build directory appears to work (build, install, and symlinks successful), but when attempting to install `rocm-hip-sdk`, I am still required to install hipblas. Apologies if this is a simple question.

### PR DESCRIPTION
**Pull request by 7910f6ba7ee4**
- update pkgver
- update checksum
- add upstream patch to fix #800

### CODE REVIEW

File: hipblas/PKGBUILD [hunk 1]
```diff
@@ -10,10 +10,18 @@ license=('MIT')
 depends=('hip' 'rocblas' 'rocsolver')
 makedepends=('cmake' 'gcc-fortran')
 _git='https://github.com/ROCmSoftwarePlatform/hipBLAS'
-source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz")
-sha256sums=('f0fdaa851971b41b48ec2e7d640746fbd6f9f433da2020c5fd95c91a7473d9e1')
+source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
+"include-path.patch::https://github.com/ROCmSoftwarePlatform/hipBLAS/commit/84b593c7c6acf176b21841a0441801d5a254c62b.patch")
```
Comment (acxz) [inline hipblas/PKGBUILD:14]: can u fix the indentation here?

File: hipblas/PKGBUILD [hunk 2]
```diff
@@ -10,10 +10,18 @@ license=('MIT')
...
+sha256sums=('5e9091dc4ef83896f5c3bc5ade1cb5db8e1a6afc451dbba4da19d8a7ec2b6f29'
+			      'SKIP')
```
Comment (acxz) [inline hipblas/PKGBUILD:16]: Can you add the checksum for the commit?

**Review by acxz — CHANGES_REQUESTED** [review]

**Comment by acxz** [comment]
Can you also generate the `.SRCINFO` file?

**Review by acxz — CHANGES_REQUESTED** [review]
Sweet this looks good! Can you just squash the commits as the AUR checks to see if each commit is valid.

**Comment by 7910f6ba7ee4** [comment]
should be squashed

File: hipblas/PKGBUILD [hunk 3]
```diff
@@ -10,10 +10,18 @@ license=('MIT')
 depends=('hip' 'rocblas' 'rocsolver')
 makedepends=('cmake' 'gcc-fortran')
 _git='https://github.com/ROCmSoftwarePlatform/hipBLAS'
-source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz")
-sha256sums=('f0fdaa851971b41b48ec2e7d640746fbd6f9f433da2020c5fd95c91a7473d9e1')
+source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
+	"include-path.patch::https://github.com/ROCmSoftwarePlatform/hipBLAS/commit/84b593c7c6acf176b21841a0441801d5a254c62b.patch")
+sha256sums=('5e9091dc4ef83896f5c3bc5ade1cb5db8e1a6afc451dbba4da19d8a7ec2b6f29'
+	    '34652acc94caad9cf9b22fcf7658b7c7a3fb4c1b517b3aa15e5cc35e973b4fc6')
```
Comment (acxz) [inline hipblas/PKGBUILD:16]: Can you indent this to the parentheses?

**Review by acxz — CHANGES_REQUESTED** [review]

File: hipblas/PKGBUILD [hunk 4]
```diff
@@ -10,10 +10,18 @@ license=('MIT')
 depends=('hip' 'rocblas' 'rocsolver')
 makedepends=('cmake' 'gcc-fortran')
 _git='https://github.com/ROCmSoftwarePlatform/hipBLAS'
-source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz")
-sha256sums=('f0fdaa851971b41b48ec2e7d640746fbd6f9f433da2020c5fd95c91a7473d9e1')
+source=("$pkgname-$pkgver.tar.gz::$_git/archive/rocm-$pkgver.tar.gz"
+		"include-path.patch::https://github.com/ROCmSoftwarePlatform/hipBLAS/commit/84b593c7c6acf176b21841a0441801d5a254c62b.patch")
+sha256sums=('5e9091dc4ef83896f5c3bc5ade1cb5db8e1a6afc451dbba4da19d8a7ec2b6f29'
+	    	'34652acc94caad9cf9b22fcf7658b7c7a3fb4c1b517b3aa15e5cc35e973b4fc6')
```
Comment (acxz) [inline hipblas/PKGBUILD:16]: This is still not indented properly.

**Review by acxz — CHANGES_REQUESTED** [review]

File: hipblas/PKGBUILD (hunk 4 above)
Comment (7910f6ba7ee4) [reply hipblas/PKGBUILD:16]: Can you elaborate? Currently the quotation are lined up like in the rocm-info PKGBUILD, but do you want the quotation marks to line up with the parentheses? Or are you referring to the misaligned spaces that show up when viewing it on the github website but not in a text editor?

Comment (acxz) [reply hipblas/PKGBUILD:16]: > Or are you referring to the misaligned spaces that show up when viewing it on the github website but not in a text editor?

The github website is showing the number of spaces just fine. I'm afraid you may have a setting such as 1 tab = 2 spaces or something unconventional. Please check your text editor settings again, until we see the proper diff. You can also check the diff local by typing in `git diff` or after the commit: `git diff HEAD~1`.

**Comment by 7910f6ba7ee4** [comment]
Used spaces instead of tabs, hopefully this works. Thanks for bearing with me through this!

**Comment by tpkessler** [comment]
Yes, now they are aligned. Thank you!

File: hipblas/.SRCINFO [hunk 5]
```diff
@@ -1,6 +1,6 @@
 pkgbase = hipblas
 	pkgdesc = ROCm BLAS marshalling library
-	pkgver = 5.1.3
+	pkgver = 5.2.0
 	pkgrel = 1
 	url = https://rocmdocs.amd.com/en/latest/ROCm_Libraries/ROCm_Libraries.html#hipblas
```
Comment (tpkessler) [inline hipblas/.SRCINFO:5]: Please update the URL to
https://hipblas.readthedocs.io/en/latest/

File: hipblas/PKGBUILD [hunk 6]
```diff
@@ -10,10 +10,18 @@ license=('MIT')
...
 
+prepare() {
+    # See https://github.com/ROCmSoftwarePlatform/hipBLAS/commit/84b593c7c6acf176b21841a0441801d5a254c62b
+    cd "$_dirname"
+    patch -Np1 -i "$srcdir/include-path.patch"
+}
+
 build() {
   # -fcf-protection is not supported by HIP, see
   # https://github.com/ROCm-Developer-Tools/HIP/blob/rocm-5.1.x/docs/markdown/clang_options.md
```
Comment (tpkessler) [inline hipblas/PKGBUILD:27]: New link is
https://docs.amd.com/bundle/ROCm-Compiler-Reference-Guide-v5.2/page/Appendix_A.html

**Comment by tpkessler** [comment]
LGTM! Thanks.

**Review by acxz — APPROVED** [review]

**Review by tpkessler — APPROVED** [review]

**Pull request merged**